datalad/support/gitrepo.py

```diff
@@ -940,10 +940,16 @@ def clone(cls, url, path, *args, **kwargs):
         if isinstance(url_ri, PathRI):
             path = Path(url)
             if not path.is_absolute():
+                # get git-created path
+                git_url = gr.config.get('remote.origin.url')
+                # Note: Not sure, whether there are circumstances where this is relative already
+                if op.isabs(git_url):
+                    # ... and make it a relative one
+                    # Note: Using os.path here, since pathlib's relative_to isn't what you'd expect
+                    git_url = op.relpath(git_url, gr.path)
+                path = Path(git_url)
                 # always in POSIX even on windows
                 path = path.as_posix()
```

Comment:
If my comment above is implemented, this conversion to and from Path is superfluous.